activerecord/test/cases/relation/where_test.rb

```diff
@@ -458,6 +458,11 @@ def test_where_on_association_with_select_relation
       assert_equal essays(:david_modest_proposal), essay
     end
 
+    def test_where_on_association_with_collection_polymorphic_relation
+      treasures = Treasure.where(name: ["diamond", "emerald"], price_estimates: PriceEstimate.all)
```

Comment:
Now we have a `Car` and `Treasure` with the same id. Before this fix it was returning false treasures with `price_estimates`.